inc/integrations/class-google-tag-manager.php

```diff
@@ -65,4 +70,29 @@ public function render_container_id_input() {
 			<input type="text" name="irving_integrations[<?php echo esc_attr( 'gtm_container_id' ); ?>]" value="<?php echo esc_attr( $gtm_key ); ?>" />
 		<?php
 	}
+
+	/**
+	 * Get data layer info for a page load.
+	 *
+	 * @param array $options Array of integration options.
+	 * @return array An array of options.
+	 */
+	public function get_data_layer( $options ) {
+		/**
+		 * Filters the data layer provided to GTM for each page.
+		 *
+		 * @param array $data_layer The data layer arguments for GTM.
+		 */
+		$data_layer = apply_filters(
+			'wp_irving_gtm_data',
+			[
+				'title' => wp_title( null, false ),
+				'url'   => get_permalink() ?: home_url(),
```

Comment:
I think this would only cover `is_single()` and `is_home()` queries for URL purposes, should we hold this until we have a drop-in function that can handle all cases?

Reply:
Good catch on this. I think we can just reuse the path passed to the REST API endpoint rather than  inferring the URL from the query. Fixed in 9b6c950